src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -176,15 +180,46 @@ private async Task RunLoop(CancellationToken cancellationToken)
                     build => azureServer.GetTimelineAsync(project, build.Id, cancellationToken)
                 );
 
+            // for each timeline, look for previousAttempts.timelineId timelines
+            // For each timeline, look at timeline.lastChangedOn for ingestion cutoff
+            // timeline.lastChangedOn will be before build.finishTime
+
             await Task.WhenAll(tasks.Select(s => s.Value));
+
+            // Look for retried timelines
+            List<(Build build, Task<Timeline> timelineTask)> retriedTimelineTasks = new List<(Build, Task<Timeline>)>();
+
+            foreach ((Build build, Task<Timeline> timelineTask) in tasks)
+            {
+                Timeline timeline = await timelineTask;
+
+                IEnumerable<string> additionalTimelineIds = timeline.Records
+                    .SelectMany(record => record.PreviousAttempts)
+                    .Where(attempt => !(attempt is null))
```

Comment:
`!(thing is null)` is a pain to read. :-)
```suggestion
                    .Where(attempt => attempt != null)
```

Reply:
I'm looking forward to being able to take C# 9's [is not null](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#pattern-matching-enhancements) 😃